accounts/abi/abi_test.go

```diff
@@ -763,6 +764,41 @@ func TestUnpackEvent(t *testing.T) {
 	}
 }
 
+// Testset (ABI and hexdata) was created based on this contract format.
+/*
+contract T {
+     event eventInDynamicType(uint elem1, string[3] elem2);
+     constructor() {}
+     function test123() public {
+         string[3] memory vals = ["A...","B...","C..."];
+         emit eventInDynamicType(123, vals);
+     }
+ }
+*/
+func TestUnpackEventOffsetBound(t *testing.T) {
+	const abiJSON = `[{"inputs":[],"stateMutability":"nonpayable","type":"constructor"},{"anonymous":false,"inputs":[{"indexed":false,"internalType":"uint256","name":"elem1","type":"uint256"},{"indexed":false,"internalType":"string[3]","name":"elem2","type":"string[3]"}],"name":"ev123","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"internalType":"uint256","name":"elem1","type":"uint256"},{"indexed":false,"internalType":"string[3]","name":"elem2","type":"string[3]"}],"name":"eventInDynamicType","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"internalType":"address","name":"sender","type":"address"},{"indexed":false,"internalType":"uint256","name":"amount","type":"uint256"},{"indexed":false,"internalType":"bytes","name":"memo","type":"bytes"}],"name":"received","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"internalType":"address","name":"sender","type":"address"}],"name":"receivedAddr","type":"event"},{"inputs":[],"name":"test123","outputs":[],"stateMutability":"nonpayable","type":"function"}]`
+
+	abi, err := JSON(strings.NewReader(abiJSON))
+	assert.Nil(t, err, err)
+
+	const rawData = `000000000000000000000000000000000000000000000000000000000000007b0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000140000000000000000000000000000000000000000000000000000000000000004141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141410000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000354242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242000000000000000000000000000000000000000000000000000000000000000000000000000000000000684343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343434343000000000000000000000000000000000000000000000000`
+	// Modify offset value (0x40 -> 0xffff) to attemp out-of-bounds access
+	modifiedRawData := rawData[:124] + "ffff" + rawData[128:]
+
+	data, err := hex.DecodeString(modifiedRawData)
+	assert.Nil(t, err, err)
```

Comment:
The 3 parameter looks ignored and L799 as well.
```go
assert.Nil(t, err) 
```

Reply:
@hyunsooda You don't have to pass `err` as the auxiliary message because when the assertion fails, it says 
'Expected nil, but got error "(error message)"'

Reply:
How about leaving L799 unchanged and instead making modifications only to L789? Because L799 expects an error.

Reply:
looks good